### PR TITLE
INGK-823 Add scan_slaves_info method

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

This PR adds the `scan_slaves_info` method that performs a scan and returns the product code and revision number of each drive.

Fixes # INGK-823

### Type of change

Please add a description and delete options that are not relevant.

- [x] Add `scan_slaves_info` method

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Build documentation locally to verify changes.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: ingenialink/canopen/network.py [hunk 1]
```diff
@@ -239,6 +244,42 @@ def scan_slaves(self) -> List[int]:
...
+            try:
+                self._setup_connection()
+            except ILError:
+                self._teardown_connection()
+                return OrderedDict()
+
+        if self._connection is None:
+            return OrderedDict()
+
+        slave_info = OrderedDict()
```
Comment (MartinALL91) [inline ingenialink/canopen/network.py:269]: You could do this at the start and then return it if something fails.

**Review by MartinALL91 — APPROVED** [review]

File: ingenialink/ethercat/network.py [hunk 2]
```diff
@@ -125,6 +125,26 @@ def scan_slaves(self) -> List[int]:
...
+
+        Raises:
+            ILError: If any slave is already connected.
+
+        """
+        try:
+            slaves = self.scan_slaves()
+        except ILError:
+            return OrderedDict()
+        slave_info = OrderedDict()
```
Comment (MartinALL91) [inline ingenialink/ethercat/network.py:142]: Same as https://github.com/ingeniamc/ingenialink-python/pull/377/files#r1496004116

File: tox.ini [hunk 3]
```diff
@@ -33,6 +33,7 @@ commands = black --check {posargs:ingenialink tests}
 description = run type checks
 skip_install = true
 deps =
+    types-dataclasses==0.6.6
```
Comment (GullinBustin) [inline tox.ini:36]: Why did you add this? Ingenialink now works with python 3.9 or higher, and dataclasses are already included. It is because of rack?

File: ingenialink/canopen/network.py [hunk 4]
```diff
@@ -239,6 +244,42 @@ def scan_slaves(self) -> List[int]:
...
+        """Scans for nodes in the network and return an ordered dict with the slave information.
+
+        Returns:
+            Ordered dict with the slave information.
+
+        """
+        try:
+            slaves = self.scan_slaves()
+        except ILError:
+            return OrderedDict()
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:257]: Why did you use OrderedDict instead of a regular dict?

Comment (pzinemanas) [reply ingenialink/canopen/network.py:257]: To ensure that the drives are correctly ordered in the Dict. So we could potentially use the keys of the dictionary as a list of scanned slaves.

File: ingenialink/canopen/network.py [hunk 5]
```diff
@@ -239,6 +244,42 @@ def scan_slaves(self) -> List[int]:
...
+            node = self._connection.add_node(slave_id)
+            product_code = convert_bytes_to_dtype(
+                node.sdo.upload(self.DRIVE_INFO_INDEX, self.PRODUCT_CODE_SUB_IX), REG_DTYPE.U32
+            )
+            revision_number = convert_bytes_to_dtype(
+                node.sdo.upload(self.DRIVE_INFO_INDEX, self.REVISION_NUMBER_SUB_IX), REG_DTYPE.U32
+            )
+            slave_info[slave_id] = SlaveInfo(int(product_code), int(revision_number))
+
+        self._teardown_connection()
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:280]: What happens if you are already connected to a drive and scan? You lose the connection, right?

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingenialink/canopen/network.py (hunk 5 above)
Comment (pzinemanas) [reply ingenialink/canopen/network.py]: Done [cbc4306](https://github.com/ingeniamc/ingenialink-python/pull/377/commits/cbc430614679d6f2a8bc1caf8807e8f0369c2684)

**Review by GullinBustin — APPROVED** [review]
